app/controllers/hosts_controller.rb

```diff
@@ -236,7 +236,7 @@ def bmc
       bmc_proxy: @host.bmc_proxy,
     }
   rescue Foreman::BMCFeatureException
-    render partial: 'bmc_missing_proxy', locals: { bmc_proxy: @host.subnet.bmc, bmc_count: SmartProxy.with_features('BMC').count }
+    render partial: 'bmc_missing_proxy', locals: { bmc_proxy: @host&.subnet&.bmc, bmc_count: SmartProxy.with_features('BMC')&.count }
```

Comment:
Interesting that in the body we use `@host.bmc_proxy` (which can actually fail with calling `nil.proxy`) but here we use `@host.subnet.bmc`. Out of interest, why do we have this difference? Shouldn't it be the same?

Reply:
I did not write this, no idea.

From the code it looks like `subnet.bmc` or `bmc_proxy` return Proxy model however `proxy` returns ProxyAPI class. These are two different things named strangely.